### PR TITLE
adding more prompts and example text to improve results

### PR DESCRIPTION
**Pull request by diatomsRcool**
Added a few more bits of text (about lions) and a few more prompts

### CODE REVIEW

File: src/.DS_Store
Comment (cmungall) [inline src/.DS_Store:1]: I think you added this accidentally

**Comment by caufieldjh** [comment]
These new examples are helpful because they reveal what happens when the input text isn't entirely specific about species - GPT refuses to work or OntoGPT assigns the non-taxon `NCBITaxon:32644`:

```
$ ontogpt extract -t biotic_interaction.Container -i tests/input/cases/biotic_interaction-012.txt
 
ERROR:root:Line 'There are no specific entities mentioned in this text. It just lists different types of interactions between organisms without specifying which taxa are involved.' does not contain a colon; ignoring
input_text:
...
prompt: |+
  From the text below, extract the following entities in the following format:

  source_taxon: <the taxon that is the subject of the interaction>
  target_taxon: <the taxon that is the object of the ineteraction>
  interaction_type: <the type of interaction>


  Text:
  compete, prey, carrion, ignore, seize, steal, challenge, mob, dominate, kill, predate, avoid, take refuge, prey on, charge at, lose kills, venture onto land, kill and eat

  ===

extracted_object: {}
```

and another example:
```
$ ontogpt extract -t biotic_interaction.Container -i tests/input/cases/biotic_interaction-011.txt
 
WARNING:root:Could not find any mappings for ECOCORE:00000102
WARNING:root:Could not find any mappings for ECOCORE:00000102
WARNING:root:Could not find any mappings for ECOCORE:00000103
WARNING:root:Could not find any mappings for ECOCORE:00000103
input_text: |-
  The lion is a generalist hypercarnivore and is considered to be both an apex and keystone predator due to its wide prey spectrum.[100][101] Its prey consists mainly of mammals, particularly ungulates weighing 190–550 kg (420–1,210 lb) with a preference for blue wildebeest, plains zebra, African buffalo, gemsbok and giraffe. Lions also hunt common warthog depending on availability, although the species is below the preferred weight range.[102] In India, sambar deer and chital are the most commonly recorded wild prey,[46][102][103] while domestic livestock may contribute significantly to their diet.[103] They usually avoid fully grown adult elephants, rhinoceroses and hippopotamus and small prey like dik-dik, hyrax, hare and monkey.[102][104] Unusual prey include porcupines and small reptiles. Lions kill other predators such as leopard, cheetah and spotted hyena but seldom consume them.[105]

  Young lions first display stalking behaviour at around three months of age, although they do not participate in hunting until they are almost a year old and begin to hunt effectively when nearing the age of two.[106] Single lions are capable of bringing down zebra and wildebeest, while larger prey like buffalo and giraffe are riskier.[90] In Chobe National Park, large prides have been observed hunting African bush elephants up to around 15 years old in exceptional cases, with the victims being calves, juveniles, and even subadults.[107][108] In typical hunts, each lioness has a favoured position in the group, either stalking prey on the "wing", then attacking, or moving a smaller distance in the centre of the group and capturing prey fleeing from other lionesses. Males attached to prides do not usually participate in group hunting.[109] Some evidence suggests, however, that males are just as successful as females; they are typically solo hunters who ambush prey in small bushland.[110]

  Lions are not particularly known for their stamina; for instance, a lioness' heart comprises only 0.57% of her body weight and a male's is about 0.45% of his body weight, whereas a hyena's heart comprises almost 1% of its body weight.[111] Thus, lions run quickly only in short bursts at about 48–59 km/h (30–37 mph) and need to be close to their prey before starting the attack.[112] One study in 2018 recorded a lion running at a top speed of 74.1 km/h (46.0 mph).[113] They take advantage of factors that reduce visibility; many kills take place near some form of cover or at night.[114] The lion's attack is short and powerful; they attempt to catch prey with a fast rush and final leap. They usually pull it down by the rump and kill by a strangling bite to the throat. They also kill prey by enclosing its muzzle in their jaws.[115]

  Lions typically consume prey at the location of the hunt but sometimes drag large prey into cover.[116] They tend to squabble over kills, particularly the males. Cubs suffer most when food is scarce but otherwise all pride members eat their fill, including old and crippled lions, which can live on leftovers.[90] Large kills are shared more widely among pride members.[117] An adult lioness requires an average of about 5 kg (11 lb) of meat per day while males require about 7 kg (15 lb).[118] Lions gorge themselves and eat up to 30 kg (66 lb) in one session;[80] if it is unable to consume all of the kill, it rests for a few hours before continuing to eat. On hot days, the pride retreats to shade with one or two males standing guard.[116] Lions defend their kills from scavengers such as vultures and hyenas.[90]

  Lions scavenge on carrion when the opportunity arises; they scavenge animals dead from natural causes such as disease or those that were killed by other predators. Scavenging lions keep a constant lookout for circling vultures, which indicate the death or distress of an animal.[119] Most carrion on which both hyenas and lions feed upon are killed by hyenas rather than lions.[56] Carrion is thought to provide a large part of lion diet.[120]
raw_completion_output: 'interactions: predation; hunting; scavenging'
prompt: |+
  Split the following piece of text into fields in the following format:

  source_taxon: <the taxon that is the subject of the interaction>
  target_taxon: <the taxon that is the object of the ineteraction>
  interaction_type: <the type of interaction>


  Text:
  scavenging

  ===

extracted_object:
  interactions:
    - source_taxon: NCBITaxon:32644
      target_taxon: NCBITaxon:32644
      interaction_type: AUTO:predation
    - source_taxon: NCBITaxon:32644
      target_taxon: NCBITaxon:32644
      interaction_type: AUTO:hunting
    - source_taxon: AUTO:N/A%20%28not%20specified%20in%20text%29
      target_taxon: AUTO:N/A%20%28not%20specified%20in%20text%29
      interaction_type: AUTO:scavenging
named_entities:
  - id: NCBITaxon:32644
    label: not specified
  - id: AUTO:predation
    label: predation
  - id: AUTO:hunting
    label: hunting
  - id: AUTO:N/A%20%28not%20specified%20in%20text%29
    label: N/A (not specified in text)
  - id: AUTO:scavenging
    label: scavenging
```

**Comment by caufieldjh** [comment]
Will try adding ECOCORE

**Comment by caufieldjh** [comment]
This helps with interaction type grounding, as expected:

(example 11)
```
extracted_object:
  interactions:
    - source_taxon: NCBITaxon:32644
      target_taxon: NCBITaxon:32644
      interaction_type: ECOCORE:00000102
    - source_taxon: NCBITaxon:32644
      target_taxon: NCBITaxon:32644
      interaction_type: AUTO:hunting
    - source_taxon: AUTO:N/A%20%28not%20specified%20in%20text%29
      target_taxon: AUTO:N/A%20%28not%20specified%20in%20text%29
      interaction_type: ECOCORE:00000103
named_entities:
  - id: NCBITaxon:32644
    label: not specified
  - id: ECOCORE:00000102
    label: predation
  - id: AUTO:hunting
    label: hunting
  - id: AUTO:N/A%20%28not%20specified%20in%20text%29
    label: N/A (not specified in text)
  - id: ECOCORE:00000103
    label: scavenging
```

**Comment by caufieldjh** [comment]
Looking into how to use these as taxon annotators:
* Catalogue of Life (COL)
* Integrated Taxonomic Information System (ITIS)
* Global Biodiversity Information Facility (GBIF) - backbone taxonomy here as darwin core archive https://hosted-datasets.gbif.org/datasets/backbone/current/backbone.zip

**Comment by diatomsRcool** [comment]
it seems to not be picking up the common names. Would you agree?

**Comment by caufieldjh** [comment]
Definitely - it's having difficulty consistently grounding anything that's not explicitly binomial. 
On a whim I tried using [OMIT](https://bioportal.bioontology.org/ontologies/OMIT) as an annotator and id prefix since it contains a simple taxonomy with common names at labels ([like the only child terms of Panthera are "Lions" and "Tigers"](https://bioportal.bioontology.org/ontologies/OMIT/?p=classes&conceptid=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FOMIT_0009130&jump_to_nav=true)) but there's a lot of non-taxonomy material in there too. Didn't notice any obvious improvement.

COL may help if I can get in in the right format (e.g., one containing both scientific and common names)

**Comment by diatomsRcool** [comment]
EOL also has tons of common names

**Comment by caufieldjh** [comment]
I've been trying to find versions (or make versions) of EOL and COL that OAK can parse, with little luck so far (most are in Darwin Core)

**Comment by diatomsRcool** [comment]
What format does OAK need?

**Comment by caufieldjh** [comment]
OAK will happily load local files in OWL, OBO, OBO Graph, or RDF.

**Comment by cmungall** [comment]
It will also be happy with anything in any OntoPortal endpoint (BioPortal,
EcoPortal, AgroPortal, ...), or an OLS instance. SKOS should also be easy
to work with.

**Comment by caufieldjh** [comment]
Will merge this before it drifts too much - will start new PR to iterate on this template + extraction further.

**Pull request merged**